apps/meteor/app/livechat/server/api/v1/visitor.ts

```diff
@@ -68,16 +68,42 @@ API.v1.addRoute('livechat/visitor', {
 			);
 		}
 
-		if (customFields && Array.isArray(customFields)) {
-			for await (const field of customFields) {
-				const customField = await LivechatCustomField.findOneById(field.key);
-				if (!customField) {
-					continue;
-				}
-				const { key, value, overwrite } = field;
-				if (customField.scope === 'visitor' && !(await VisitorsRaw.updateLivechatDataByToken(token, key, value, overwrite))) {
-					return API.v1.failure();
-				}
+		if (customFields && Array.isArray(customFields) && customFields.length > 0) {
+			const keys = customFields.map((field) => field.key);
+			const errors: string[] = [];
+
+			const processedKeys = await Promise.all(
```

Comment:
Do we need to have this Promise.all?

Reply:
Sadly yes because of the .map on the cursor 😬

Reply:
But we are already `await`ing it (`await LivechatCustomField.findByIdsAndScope`), there is no need for the Promise.all

Reply:
Interesting. I tested and the async map always returned a Promise[] 🤔 

but apparently it works both ways lol

Reply:
I pushed and rememberd why:
`.map.toArray` returns a `Promise<X>` type, when you await, then you get `X`
In this case, since the `map` is doing an async operation, the result `X` is of type `Promise<string | undefined>[]`.

That means, we do get an array of promises 😬 because of the map. It apparently works if we ignore the types, which is weird. I got the same behavior and everything apparently worked. BUT, TS won't allow it cause it's actually a promise array.

I'll leave it like that for now